### PR TITLE
Add note to building.md for avoid build error which caused by proto-s…

### LINKED ISSUE #35846: Write a way  in building.md to avoid build failures that occur due to the proto-schema-compatibility-maven-plugin check in a proxy environment
### Description

The proto-schema-compatibility-maven-plugin check added by the following issue may cause build failures in a proxy environment.
[Add proto-schema-compatibility-maven-plugin check against release/26.0 branch to main · Issue #33565 · keycloak/keycloak](https://github.com/keycloak/keycloak/issues/33565)　
To avoid this build failure, you can skip the check by adding a property as shown in the comments here.
https://github.com/keycloak/keycloak/issues/33565#issuecomment-2522659984
We should write down this know-how in building.md.

### PR DESCRIPTION
**Pull request by hrttthr**
…chema-compatibility-maven-plugin check

Closes #35846

### CODE REVIEW

**Review by tnorimat — APPROVED** [review]
LGTM

**Comment by tnorimat** [comment]
@ryanemerson @mposolda Hello, this PR incorporate [the comment](https://github.com/keycloak/keycloak/issues/33565#issuecomment-2522659984). IMO, it is worth mentioning the comment on the README, so the PR could be merged. WDYT?

File: docs/building.md [hunk 1]
```diff
@@ -47,6 +47,15 @@ To enable it by default, add it to the `MAVEN_OPTS` environment variable:
 
     export MAVEN_OPTS="-Dmaven.build.cache.enabled=true"
 
+---
+**NOTE**
+
+To ensure that development in a branch does not break compatibility with existing releases, proto-schema-compatibility-maven-plugin checks may be run, which can cause builds to fail in proxy environments.
+To avoid this, you can skip this check by adding the following property:
+    -DskipProtoLock=true
```
Comment (ahus1) [inline docs/building.md:55]: To render this as a pre-formatted code block, a blank line is necessary in Markdown.  

```suggestion

    -DskipProtoLock=true
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank you for the PR. Please update the formatting of the changes so that a code block is rendered. Thanks!

**Review by ryanemerson — APPROVED** [review]
Thanks for the contribution, LGTM once Alexander's comment has been addressed.

**Review by ahus1 — APPROVED** [review]

**Pull request merged**